### DISCUSSION TITLE
Copilot suggests obscenities

### DISCUSSION THREAD

**Discussion by fquffio** [community/community#8304] [Copilot Conversations]
I mistyped "documents" and started writing "dioc". Copilot promptly suggested "diocane" as a completion, an Italian swearing which consists in basically saying "God is a dog" — an obscenity comparable to the F-word in English, and possibly even worse. It would quite surely get someone fired immediately if they say it on TV, just to give you the idea, perhaps even with a public apology, and never appears in movies, songs nor in a videogame.

I personally had a laugh when this happened, but just in case someone may feel disturbed, you might want to consider excluding this expression from Copilot's vocabulary — assuming you have such a list of excluded words/expressions.

Just for the sake of completeness, the expression would actually consist of two separate words (dio c***), but it is often written as one. And just like swearings in any other language that I personally know of, can have plenty of variants.

![GitHub Copilot suggestion: Error deleting diocane object from index.](https://user-images.githubusercontent.com/7108146/144253352-7dc7d6c8-4275-4d10-b0ad-4ace4eb25378.png)

**Comment by Urethramancer** [comment]
It also seems to be a beer of some sort ;)

Plenty of English would be censored if you apply other languages' swarasauruses to it, so I guess the correct way to fix this is to make a list of misspellings and their best corrections. That would probably be a very long-term goal, as it has to be crowdsourced.

**Comment by cfBrianMiller** [comment]
+1 To this I got this as a suggested comment line

\#F*** it, i'll use a switch statement

This was in powershell
